doc/index.md

```diff
@@ -113,4 +113,41 @@ public class Connection
         var myResult = await rpc.InvokeWithParameterObjectAsync<string>("test/InvokeTestMethod");
     }
 }
+```
+
+## Crashing the process on exception
+In some cases, you may want to immediately crash the server and client processes if certain exceptions are thrown. In this case, overriding the `IsFatalException` method will give you the desired functionality. `IsFatalException` provides an exception filter through which you can access and respond to exceptions as they are caught.
+```csharp
+public class Server : BaseClass
+{
+    public void ThrowsException() => throw new Exception("Throwing an exception");
+}
+
+public class JsonRpcCrashesOnException : JsonRpc
+{
+    public JsonRpcCrashesOnException(Stream clientStream, Stream serverStream, object target = null) : base(clientSteam, serverStream, target)
+    {
+    }
+
+    protected override bool IsFatalException(Exception ex)
+    {
+        if (ex.GetType() != typeof(OperationCanceledException))
+        {
+            Environment.FailFast(ex.message);
+        }
+
+        return false;
+    }
+}
+
+public class Connection
+{
+    public async Task NotifyRemote()
+    {
+        var target = new Server();
+        var rpc = new JsonRpcCrashesOnException(Console.OpenStandardOutput(), Console.OpenStandardInput(), target);
```

Comment:
I'm thoroughly confused by this. It appears you're working with *console* streams, which is a little odd in the first place. But then you're using the *same* instance of JsonRpc both to invoke the server method as a client and also to invoke it as the server itself. I'm pretty sure that doesn't work.

Reply:
I'm mostly trying to stay consistent with the rest of the documentation (although I see another example of using the constructor directly uses `FullDuplexStream`). I do agree with your confusion though; these examples confused me when I first started looking into this and I got more value from looking at the unit tests.

Reply:
Ok. I didn't thoroughly review the docs we have right now, which came from Tina. So feel free to improve on them in your addition, and possibly fix up the originals too. :)